core/lib/types/src/fee_model.rs

```diff
@@ -236,9 +237,76 @@ pub struct FeeParamsV1 {
 
 #[derive(Debug, Clone, Copy, Serialize, Deserialize)]
 pub struct FeeParamsV2 {
-    pub config: FeeModelConfigV2,
-    pub l1_gas_price: u64,
-    pub l1_pubdata_price: u64,
+    config: FeeModelConfigV2,
+    l1_gas_price: u64,
+    l1_pubdata_price: u64,
+    conversion_ratio: BaseTokenConversionRatio,
+}
+
+impl FeeParamsV2 {
+    pub fn new(
+        config: FeeModelConfigV2,
+        l1_gas_price: u64,
+        l1_pubdata_price: u64,
+        conversion_ratio: BaseTokenConversionRatio,
+    ) -> Self {
+        Self {
+            config,
+            l1_gas_price,
+            l1_pubdata_price,
+            conversion_ratio,
+        }
+    }
+
+    /// Returns the fee model config with the minimal L2 gas price denominated in the chain's base token (WEI or equivalent).
+    pub fn config(&self) -> FeeModelConfigV2 {
+        FeeModelConfigV2 {
+            minimal_l2_gas_price: self.convert_to_base_token(self.config.minimal_l2_gas_price),
```

Comment:
If you do that, we need also to change the logic in the smart contract

https://github.com/matter-labs/era-contracts/blob/db9387690502937de081a959b164db5a5262ce0a/l1-contracts/contracts/state-transition/chain-deps/facets/Mailbox.sol#L180

otherwise it will become very confusing

Reply:
@thomas-nguy I think we covered this offline that no SC changes are needed. If you need more context or disagree please unresolve